package Exception;

public class ExceptionHandling {

	public static void main(String[] args) {
		// TODO Auto-generated method stub
		System.out.println("first statement");
		
		int arr[] = {1,2,3,4};
		try {
		System.out.println(arr[1]);
		String str = "Anjali";
		System.out.println(str.charAt(4));
		System.out.println(str.charAt(8));
		//exception occured here
		System.out.println(arr[2]);//will not run
		}
		catch(ArrayIndexOutOfBoundsException e) {//exception is the parent if all types of exeption
			System.out.println(e.toString());
			System.out.println(e.getMessage());
			System.out.println(e.getStackTrace());
		}
		catch(StringIndexOutOfBoundsException e) {//exception is the parent if all types of exeption
			System.out.println(e.toString());
			System.out.println(e.getMessage());
			System.out.println(e.getStackTrace());
		}
		
		System.out.println("another statement");

	}

}
